### PR TITLE
User event application

### PR DESCRIPTION
**Pull request by goharahmed**
Allows ARI to trigger AMI layer userevent. Please provide feedback. Changes corresponding to this in ari-proxy to be followed shortly in the respective repository. 

Thank you.

Ref:
https://wiki.asterisk.org/wiki/display/AST/Asterisk+18+Events+REST+API#Asterisk18EventsRESTAPI-userEvent

- - -
This change is [Reviewable](https://reviewable.io/reviews/cycoresystems/ari/139)

### CODE REVIEW

**Comment by Ulexus** [comment]
Nice, thanks!

**Comment by Ulexus** [comment]
It looks like you're failing the linter with formatting issues.  Can you be sure to run through `gofumpt`?

**Comment by goharahmed** [comment]
Thank you so much for quick review. 
Its linted by VSCode, as Im not familiar with `gofumpt`  
Do you want me to run this: 
```gofumpt -l -w .```
from this library: https://github.com/mvdan/gofumpt

**Comment by Ulexus** [comment]
Also:  https://golangci-lint.run/usage/integrations/

**Comment by Ulexus** [comment]
Ah, I see; it must be a newer linter restriction since the last commit.
If you don't mind fixing it here, that would be great.

**Comment by goharahmed** [comment]
Let me see as how to fix it :)

**Comment by Ulexus** [comment]
Should just be to replace with https://pkg.go.dev/golang.org/x/text/cases#Title

**Comment by goharahmed** [comment]
@Ulexus - I think all set, I was reading about the "Language.Und" and I don't fully comprehend what could the "x" string possible values and if the Undisclosed language rules will make it work or not. 
Anways, the PR related to ari-proxy is ready but waiting for the main ari package to show the UserEvent implementation
```
s.ari.Channel().UserEvent undefined (type ari.Channel has no field or method UserEvent)
```

Once this PR is closed I'll submit the next one.

File: internal/eventgen/main.go [hunk 1]
```diff
@@ -151,7 +154,8 @@ func main() {
 
 			items := strings.Split(pkey, "_")
 			for _, x := range items {
-				name += strings.Title(x)
+				caser := cases.Title(language.Und)
```
Comment (Ulexus) [inline internal/eventgen/main.go:157]: I think we can safely assume the Asterisk developers to stay with English event names, so we should be able to use `language.English` here.
Also, we shouldn't need the caser particularly, so you could write this as:

```golang
name += cases.Title(language.English).String(x)
```

**Comment by Ulexus** [comment]
Hmm... it looks like there's a mess of additional linter errors after that one.  I'm going to go ahead and merge this and then fix the rest myself before tagging the next release.

Thanks for your submission!

**Comment by Ulexus** [comment]
Tagged v5.3.0 with this feature, @goharahmed

**Pull request merged**